--- ./ld/ld.c.orig	2006-03-28 18:04:08.000000000 -0800
+++ ./ld/ld.c	2006-06-03 11:45:07.000000000 -0700
@@ -583,6 +583,9 @@
 
 	        p = &(argv[i][1]);
 		switch(*p){
+		case 'B': 
+			if(strcmp(p, "Bstatic") && strcmp(p, "Bdynamic")) goto unknown_flag;
+			break;
 		case 'l':
 		    if(p[1] == '\0')
 			fatal("-l: argument missing");
@@ -2693,6 +2696,16 @@
 	    else{
 		p = &(argv[i][1]);
 		switch(*p){
+		case 'B':
+			if(strcmp(p, "Bstatic") == 0){
+				search_lib_extensions[0]=".a";
+				search_lib_extensions[1]=".dylib";
+				break;
+			} else if (strcmp(p, "Bdynamic") == 0){
+				search_lib_extensions[0]=".dylib";
+				search_lib_extensions[1]=".a";
+				break;
+			}
 		case 'b':
 		    if(strcmp(p, "bundle_loader") == 0){
 			/*
--- ./ld/pass1.c.orig	2006-03-28 18:04:16.000000000 -0800
+++ ./ld/pass1.c	2006-06-03 11:45:07.000000000 -0700
@@ -128,6 +128,13 @@
 
 /* the pointer to the head of the base object file's segments */
 __private_extern__ struct merged_segment *base_obj_segments = NULL;
+
+__private_extern__ char *search_lib_extensions[] = {
+	".dylib",
+	".a",
+	NULL
+};
+
 #endif /* !defined(RLD) */
 
 #if !defined(SA_RLD) && !(defined(KLD) && defined(__STATIC__))
@@ -344,6 +351,7 @@
     char *file_name;
 #ifndef RLD
     char *p, *type;
+    int search_lib_index = 0;
 #endif /* !defined(RLD) */
     kern_return_t r;
     unsigned long file_size;
@@ -361,7 +369,6 @@
 	/* this function" can safely be ignored */
 	file_name = NULL;
 #endif /* DEBUG */
-
 	fd = -1;
 #ifndef RLD
 	if(lname){
@@ -383,12 +390,11 @@
 			search_paths_for_lname(&name[2], &file_name, &fd);
 		    }
 		    else{
-			p = mkstr("lib", &name[2], ".dylib", NULL);
-			search_for_file(p, &file_name, &fd);
-			if(fd == -1){
-			    p = mkstr("lib", &name[2], ".a", NULL);
-			    search_for_file(p, &file_name, &fd);
-			}
+			while (search_lib_extensions[search_lib_index] && (fd == -1)) {
+				p = mkstr("lib", &name[2], search_lib_extensions[search_lib_index], NULL);
+				search_for_file(p, &file_name, &fd);
+				search_lib_index++;
+				}
 		    }
 		}
 		else{
@@ -641,15 +647,16 @@
 char **file_name,
 int *fd)
 {
-	*file_name = mkstr(dir, "/", "lib", lname_argument, ".dylib", NULL);
-	if((*fd = open(*file_name, O_RDONLY)) != -1)
-	    return;
-	free(*file_name);
-
-	*file_name = mkstr(dir, "/", "lib", lname_argument, ".a", NULL);
-	if((*fd = open(*file_name, O_RDONLY)) != -1)
-	    return;
-	free(*file_name);
+	int search_lib_index=0;
+	*fd = -1;
+	while (search_lib_extensions[search_lib_index] && (*fd == -1)) {
+		*file_name = mkstr(dir, "/", "lib", lname_argument, search_lib_extensions[search_lib_index], NULL);
+		if((*fd = open(*file_name, O_RDONLY)) != -1)
+			break;
+	search_lib_index++;
+	}
+	if (*fd == -1)
+		free(*file_name);
 }
 #endif /* !defined(RLD) */
 
--- ./ld/pass1.h.orig	2006-01-27 15:20:21.000000000 -0800
+++ ./ld/pass1.h	2006-06-03 11:45:07.000000000 -0700
@@ -63,6 +63,8 @@
 /* the pointer to the head of the base object file's segments */
 __private_extern__ struct merged_segment *base_obj_segments;
 
+__private_extern__ char *search_lib_extensions[];
+
 /*
  * These are pointers to strings and symbols used to search of the table of
  * contents of a library.  These have to be can not be local so that routines
--- ./man/ld.1.orig	2006-01-27 15:20:31.000000000 -0800
+++ ./man/ld.1	2006-06-03 11:45:07.000000000 -0700
@@ -1738,6 +1738,18 @@
 For dynamic link editor files, specifies the name of the file
 the dynamic link editor will be installed in for programs that use it.
 .TP
+.BI \-Bstatic
+Search for static archives before dynamic shared libraries. Both
+.BI \-Bstatic
+and
+.BI \-Bdynamic
+can be specified multiple times, and affect only the following
+.BI \-l
+arguments.
+.TP
+.BI \-Bdynamic
+This is the default. Search for dynamic shared libraries first, then static.
+.TP
 .BI \-macosx_version_min " version"
 This overrides the
 .B MACOSX_DEPLOYMENT_TARGET
--- ./man/libtool.1.orig	2006-03-14 14:17:42.000000000 -0800
+++ ./man/libtool.1	2006-06-03 11:45:07.000000000 -0700
@@ -308,6 +308,8 @@
 .B \-m,
 .B \-dead_strip,
 .B \-no_dead_strip_inits_and_terms,
+.B \-Bstatic,
+.B \-Bdynamic,
 .B \-executable_path,
 .B \-syslibroot,
 .B \-no_uuid.
--- ./misc/libtool.c.orig	2006-03-14 14:17:47.000000000 -0800
+++ ./misc/libtool.c	2006-06-03 11:45:07.000000000 -0700
@@ -812,6 +812,15 @@
 		    cmd_flags.files[cmd_flags.nfiles++] = argv[i];
 		    lflags_seen = TRUE;
 		}
+		else if(strncmp(argv[i], "-Bstatic", 8) == 0 ||
+			    strncmp(argv[i], "-Bdynamic",9) == 0){
+		    if(cmd_flags.ranlib == TRUE){
+			error("unknown option: %s", argv[i]);
+			usage();
+		    }
+		    cmd_flags.files[cmd_flags.nfiles++] = argv[i];
+		    lflags_seen = TRUE;
+		}
 		else if(strncmp(argv[i], "-weak-l", 7) == 0){
 		    if(cmd_flags.ranlib == TRUE){
 			error("unknown option: %s", argv[i]);
@@ -1204,6 +1213,10 @@
 		i++;
 		continue;
 	    }
+		else if(strncmp(cmd_flags.files[i], "-Bstatic",8) == 0 ||
+			strncmp(cmd_flags.files[i], "-Bdynamic",9) == 0){
+		continue;
+		}
 	    else{
 		if(ofile_map(cmd_flags.files[i], NULL, NULL, ofiles + i,
 			     TRUE) == FALSE)
