--- ld/ld.c.orig	2004-11-07 22:02:38.000000000 -0800
+++ ld/ld.c	2004-11-07 22:04:07.000000000 -0800
@@ -78,6 +78,8 @@
 #include "layout.h"
 #include "pass2.h"
 
+#include <config.h>
+
 /* name of this program as executed (argv[0]) */
 __private_extern__ char *progname = NULL;
 /* indication of an error set in error(), for processing a number of errors
@@ -1795,6 +1797,11 @@
 	 * cross build environments.
 	 */
 	next_root = getenv("NEXT_ROOT");
+	if(next_root == NULL) {
+#ifdef CROSS_SYSROOT
+	  next_root = CROSS_SYSROOT;
+#endif
+	}
 	if(next_root != NULL){
 	    for(i = 0; standard_dirs[i] != NULL; i++){
 		p = allocate(strlen(next_root) +
--- misc/libtool.c.orig	2004-11-07 22:04:40.000000000 -0800
+++ misc/libtool.c	2004-11-07 22:05:14.000000000 -0800
@@ -53,6 +53,8 @@
 #else
 #endif
 
+#include <config.h>
+
 /* used by error routines as the name of the program */
 char *progname = NULL;
 
@@ -927,6 +929,11 @@
 	 * cross build environments.
 	 */
 	next_root = getenv("NEXT_ROOT");
+        if(next_root == NULL) {
+#ifdef CROSS_SYSROOT
+          next_root = CROSS_SYSROOT;
+#endif
+        }
 	if(next_root != NULL){
 	    for(i = 0; standard_dirs[i] != NULL; i++){
 		p = allocate(strlen(next_root) +
