diff -ur ../as.old/app.c ./app.c
--- ../as.old/app.c	2004-07-27 09:04:06.000000000 -0700
+++ ./app.c	2004-09-02 15:36:17.000000000 -0700
@@ -29,6 +29,10 @@
 #include "app.h"
 #include "messages.h"
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 FILE *scrub_file = NULL;
 char *scrub_string = NULL;
 char *scrub_last_string = NULL;
diff -ur ../as.old/input-file.c ./input-file.c
--- ../as.old/input-file.c	2004-07-27 09:04:06.000000000 -0700
+++ ./input-file.c	2004-09-02 15:36:30.000000000 -0700
@@ -37,6 +37,10 @@
 #include "app.h"
 #include "as.h"
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 /*
  * This variable is non-zero if the file currently being read should be
  * preprocessed by app.  It is zero if the file can be read straight in.
diff -ur ../as.old/read.c ./read.c
--- ../as.old/read.c	2004-07-30 11:51:28.000000000 -0700
+++ ./read.c	2004-09-02 15:36:40.000000000 -0700
@@ -57,6 +57,10 @@
 #include "xmalloc.h"
 #include "app.h"
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 /*
  * Parsing of input is done off of this pointer which points to the next char
  * of source file to parse.
