--- ./app.c.orig	2006-02-20 13:45:43.000000000 -0800
+++ ./app.c	2006-02-21 18:53:49.000000000 -0800
@@ -29,6 +29,10 @@
 #include "app.h"
 #include "messages.h"
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 FILE *scrub_file = NULL;
 char *scrub_string = NULL;
 char *scrub_last_string = NULL;
--- ./input-file.c.orig	2006-02-20 13:45:43.000000000 -0800
+++ ./input-file.c	2006-02-21 18:53:49.000000000 -0800
@@ -37,6 +37,10 @@
 #include "app.h"
 #include "as.h"
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 /*
  * This variable is non-zero if the file currently being read should be
  * preprocessed by app.  It is zero if the file can be read straight in.
--- ./read.c.orig	2006-02-20 13:45:43.000000000 -0800
+++ ./read.c	2006-02-21 18:53:49.000000000 -0800
@@ -57,6 +57,10 @@
 #include "xmalloc.h"
 #include "app.h"
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 /*
  * Parsing of input is done off of this pointer which points to the next char
  * of source file to parse.
