--- ./input-scrub.c.orig	2003-01-13 16:54:43.000000000 -0800
+++ ./input-scrub.c	2006-03-15 20:35:29.000000000 -0800
@@ -372,14 +372,7 @@
   fprintf (stderr,"as:file(%s) %s! ",
 	   filename, gripe
 	   );
-  if (errno > sys_nerr)
-    {
-      fprintf (stderr, "Unknown error #%d.", errno);
-    }
-  else
-    {
-      fprintf (stderr, "%s.", sys_errlist [errno]);
-    }
+      fprintf (stderr, "%s.", strerror(errno));
   (void)putc('\n', stderr);
   errno = 0;			/* After reporting, clear it. */
   if (input_file_is_open())	/* RMS says don't mention line # if not needed. */
