--- ./relax.h.orig	1993-09-09 13:09:18.000000000 -0700
+++ ./relax.h	2006-03-15 20:35:30.000000000 -0800
@@ -1,3 +1,6 @@
+#ifndef _RELAX_H_
+#define _RELAX_H_
+
 /* The type used for a target address */
 typedef unsigned long relax_addressT;
 
@@ -42,3 +45,6 @@
     relax_substateT rlx_more;	  /* Next longer relax-state. */
 				  /* 0 means there is no 'next' relax-state. */
 } relax_typeS;
+
+#endif // _RELAX_H_
+
