--- ./read.c.orig	2009-06-11 12:30:36.000000000 -0500
+++ ./read.c	2009-06-11 12:31:10.000000000 -0500
@@ -37,6 +37,7 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <inttypes.h>
 #include "stuff/round.h"
 #include "stuff/arch.h"
 #include "stuff/best_arch.h"
@@ -2505,7 +2506,7 @@
 
 	*p = 0;
 	if(strlen(segname) > sizeof(s.segname)){
-	    as_bad("segment-name: %s too long (maximum %ld characters)",
+	    as_bad("segment-name: %s too long (maximum %" PRIdPTR " characters)",
 		    segname, sizeof(s.segname));
 	    ignore_rest_of_line();
 	    *p = c;
@@ -2514,7 +2515,7 @@
 
 	*q = 0;
 	if(strlen(sectname) > sizeof(s.sectname)){
-	    as_bad("section-name: %s too long (maximum %ld characters)",
+	    as_bad("section-name: %s too long (maximum %" PRIdPTR " characters)",
 		    sectname, sizeof(s.sectname));
 	    ignore_rest_of_line();
 	    return;
@@ -2664,7 +2665,7 @@
 
 	*p = 0;
 	if(strlen(segname) > sizeof(s.segname)){
-	    as_bad("segment-name: %s too long (maximum %ld characters)",
+	    as_bad("segment-name: %s too long (maximum %" PRIdPTR " characters)",
 		    segname, sizeof(s.segname));
 	    ignore_rest_of_line();
 	    *p = c;
@@ -2673,7 +2674,7 @@
 
 	*q = 0;
 	if(strlen(sectname) > sizeof(s.sectname)){
-	    as_bad("section-name: %s too long (maximum %ld characters)",
+	    as_bad("section-name: %s too long (maximum %" PRIdPTR " characters)",
 		    sectname, sizeof(s.sectname));
 	    ignore_rest_of_line();
 	    *p = c;
