--- ./ar.c.orig	2006-04-28 18:41:59.000000000 -0500
+++ ./ar.c	2009-06-11 12:30:42.000000000 -0500
@@ -84,7 +84,7 @@
 #include "extern.h"
 #include "stuff/execute.h"
 #include "stuff/unix_standard_mode.h"
-
+#include "stuff/allocate.h"
 CHDR chdr;
 u_int options;
 char *archive, *envtmp, *posarg, *posname;
@@ -276,7 +276,7 @@
 	if(run_ranlib){
 	    /* run ranlib -f or -q on the archive */
 	    reset_execute_list();
-	    add_execute_list("ranlib");
+	    add_execute_list(makestr(BINDIR, "/", RANLIBPROG, NULL));
 	    if(options & AR_S)
 		add_execute_list("-f");
 	    else
