--- symbols.c.orig	2005-03-06 19:49:49.000000000 -0800
+++ symbols.c	2005-03-06 19:55:53.000000000 -0800
@@ -2502,6 +2502,11 @@
 		    }
 		}
 		if(resolve_flat == TRUE){
+                    if(twolevel_namespace == TRUE &&
+                         (((struct mach_header *)(cur_obj->obj_addr))->
+                         flags & MH_TWOLEVEL) != MH_TWOLEVEL){
+                        continue; /* with for loop */
+                    }
 		    /* lookup the symbol and see if it has already been seen */
 		    hash_pointer = lookup_symbol(symbol_name);
 		    if(*hash_pointer == NULL){
