--- ./ld.c.orig	2005-10-31 10:04:32.000000000 -0800
+++ ./ld.c	2005-10-31 10:04:35.000000000 -0800
@@ -53,12 +53,10 @@
 #ifndef RLD
 #include "stuff/symbol_list.h"
 #endif
-#include "make.h"
 #include <mach/mach_init.h>
 #if defined(__OPENSTEP__) || defined(__GONZO_BUNSEN_BEAKER__)
 #include <servers/netname.h>
 #else
-#include <servers/bootstrap.h>
 #endif
 #else /* defined(KLD) && defined(__STATIC__) */
 #include <mach/mach.h>
@@ -3236,6 +3234,7 @@
 void
 check_for_ProjectBuilder(void)
 {
+#if OLD_PROJECTBUILDER_INTERFACE
     char *portName;
 #if defined(__OPENSTEP__) || defined(__GONZO_BUNSEN_BEAKER__)
     char *hostName;
@@ -3259,6 +3258,7 @@
 	if(ProjectBuilder_port == MACH_PORT_NULL)
 	    return;
 	talking_to_ProjectBuilder = 1;
+#endif
 }
 
 /*
@@ -3270,6 +3270,7 @@
 tell_ProjectBuilder(
 char *message)
 {
+#if OLD_PROJECTBUILDER_INTERFACE
 	make_alert(ProjectBuilder_port,
 	    2, /* eventType */
 	    NULL, 0, /* functionName, not used by ProjectBuilder */
@@ -3277,6 +3278,7 @@
 	    NULL, 0, /* directory */
 	    0, /* line */
 	    message, strlen(message)+1 > 1024 ? 1024 : strlen(message)+1);
+#endif
 }
 
 /*
