--- ld.c.orig	2004-11-07 14:59:09.000000000 -0800
+++ ld.c	2004-11-07 15:09:48.000000000 -0800
@@ -52,12 +52,10 @@
 #ifndef RLD
 #include "stuff/symbol_list.h"
 #endif
-#include "make.h"
 #include <mach/mach_init.h>
 #if defined(__OPENSTEP__) || defined(__GONZO_BUNSEN_BEAKER__)
 #include <servers/netname.h>
 #else
-#include <servers/bootstrap.h>
 #endif
 #else /* defined(KLD) && defined(__STATIC__) */
 #include <mach/mach.h>
@@ -2923,6 +2921,7 @@
 void
 check_for_ProjectBuilder(void)
 {
+#if OLD_PROJECTBUILDER_INTERFACE
     char *portName;
 #if defined(__OPENSTEP__) || defined(__GONZO_BUNSEN_BEAKER__)
     char *hostName;
@@ -2946,6 +2945,7 @@
 	if(ProjectBuilder_port == MACH_PORT_NULL)
 	    return;
 	talking_to_ProjectBuilder = 1;
+#endif
 }
 
 /*
@@ -2957,6 +2957,7 @@
 tell_ProjectBuilder(
 char *message)
 {
+#if OLD_PROJECTBUILDER_INTERFACE
 	make_alert(ProjectBuilder_port,
 	    2, /* eventType */
 	    NULL, 0, /* functionName, not used by ProjectBuilder */
@@ -2964,6 +2965,7 @@
 	    NULL, 0, /* directory */
 	    0, /* line */
 	    message, strlen(message)+1 > 1024 ? 1024 : strlen(message)+1);
+#endif
 }
 
 /*
