--- ./libtool.c.orig	2006-02-21 18:53:56.000000000 -0800
+++ ./libtool.c	2006-02-21 18:54:01.000000000 -0800
@@ -2369,7 +2369,7 @@
 	    if((archs[i].arch_flag.cputype & CPU_ARCH_ABI64) == CPU_ARCH_ABI64)
 		add_execute_list("ld64");
 	    else
-		add_execute_list("ld");
+	      add_execute_list(makestr(BINDIR, "/", LDPROG, NULL));
 	    if(narchs != 0 && cmd_flags.arch_only_flag.name == NULL)
 		add_execute_list("-arch_multiple");
 	    if(archs != NULL){
