--- ./libtool.c.orig	2006-06-03 11:45:07.000000000 -0700
+++ ./libtool.c	2006-06-03 11:45:08.000000000 -0700
@@ -2377,7 +2377,7 @@
 	    if((archs[i].arch_flag.cputype & CPU_ARCH_ABI64) == CPU_ARCH_ABI64)
 		add_execute_list("ld64");
 	    else
-		add_execute_list("ld");
+	      add_execute_list(makestr(BINDIR, "/", LDPROG, NULL));
 	    if(narchs != 0 && cmd_flags.arch_only_flag.name == NULL)
 		add_execute_list("-arch_multiple");
 	    if(archs != NULL){
