--- ./libtool.c.orig	2009-06-11 12:30:41.000000000 -0500
+++ ./libtool.c	2009-06-11 12:30:58.000000000 -0500
@@ -51,12 +51,10 @@
 #include "stuff/lto.h"
 #endif /* LTO_SUPPORT */
 
-#include "make.h"
 #include <mach/mach_init.h>
 #if defined(__OPENSTEP__) || defined(__GONZO_BUNSEN_BEAKER__)
 #include <servers/netname.h>
 #else
-#include <servers/bootstrap.h>
 #endif
 
 /* used by error routines as the name of the program */
@@ -2828,6 +2826,7 @@
 char *arch_name,
 char *fileName)
 {
+#ifdef OLD_PROJECTBUILDER_INTERFACE
     char message_buf[1024];
     mach_port_t ProjectBuilder_port;
     char *portName;
@@ -2863,6 +2862,7 @@
 	    NULL, 0,
 	    0,
 	    message_buf, strlen(message_buf) + 1);
+#endif
 }
 
 /*
