--- libtool.c.orig	2004-09-05 16:39:51.000000000 -0700
+++ libtool.c	2004-09-05 16:40:31.000000000 -0700
@@ -273,14 +273,9 @@
 	toc_mode = S_IFREG | (0666 & ~oumask);
 	(void)umask(oumask);
 
-	/* see if this is being run as ranlib */
-	p = strrchr(argv[0], '/');
-	if(p != NULL)
-	    p++;
-	else
-	    p = argv[0];
-	if(strncmp(p, "ranlib", sizeof("ranlib") - 1) == 0)
-	    cmd_flags.ranlib = TRUE;
+#ifdef RANLIB
+	cmd_flags.ranlib = TRUE;
+#endif
 
 	/* The default is to used long names */
 	cmd_flags.use_long_names = TRUE;
