--- ./redo_prebinding.c.orig	2006-09-09 13:49:18.000000000 -0700
+++ ./redo_prebinding.c	2006-09-09 13:50:19.000000000 -0700
@@ -82,7 +82,7 @@
  */
 #include <mach-o/redo_prebinding.h>
 #endif /* defined(LIBRARY_API) */
-
+#include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -9263,7 +9263,7 @@
 	errors++;
 }
 #endif /* defined(LIBRARY_API) */
-
+#if HAVE_GETATTRLIST
 #include <sys/attr.h>
 /*
  * Structure defining what's returned from getattrlist.  It returns all the
@@ -9337,7 +9337,15 @@
 
 	return(FALSE);
 }
-
+#else
+static
+enum bool
+has_resource_fork(
+		  char *filename)
+{
+  return FALSE;
+}
+#endif
 
 static unsigned long find__dyld_section_addr(const struct mach_header* mh)
 {
